_posts/2021-01-23-Agrelon.md

```diff
@@ -0,0 +1,90 @@
+ontology](http://xmlns.com/foaf/0.1/).
+
+In the course of development of IrishGen, more predicates needed to be
+added to cover some more areas that the Relationship ontology did not.
+Additional tweaks were also made to make the `parentOf` relation a
+subproperty of the `ancestorOf` predicate (and inversely for the
+`childOf`) as this would not allow transitivity backwards (also
+remember that your parents are your ancestors who just happen to still
+be alive).
+
+In private email conversations with Lars Svensson of the Deutsche
+Nationalbibliothek, he brought to my attention that there is a serious
+flaw in the way in which the Relationship ontology uses
+`owl:equivalentClass`.  He proposed that IrishGen use the ontology for
+people and institutions which was developed there, called
+[Agrelon](https://d-nb.info/standards/elementset/agrelon).  Then as
+now, switching from using the Relationship ontology to using Agrelon
+would involve a lot of work so I politely declined.  However, I kept
+the idea in my mind.
+
+While working with Stardog, you will notice that some queries are just
+plain slow.  This was a constant irritant as I preferred Stardog's
+strict adherence to standards.  So, in a fit of aggravation, I did a
+quick and dirty transformation of IrishGen from the Relationship
+ontology to Agrelon as I remembered the flaw that Lars had pointed out.
+In the case that I was working with, I got a massive speed boost to
+the query.  This gave me a boost in attempting to integrate Agrelon
+into IrishGen.
+
+The more interested among you may have noticed that in the `utils`
+directory in the IrishGen repository has gained the Python file
+[`irish_gen_add_agrelon.py`](https://github.com/cyocum/irish-gen/blob/master/utils/irish_gen_add_agrelon.py).
+This Python script will, with the proper libraries installed (see code
+for details), add Agrelon to an IrishGen TRiG file.  A user can now
+run this against one or all of IrishGen to add Agrelon predicates to
+IrishGen.  Also, I learned a few things about Agrelon while writing
+the script.  For instance, Agrelon contains many of the relations that
+the Relationship ontology lacked and needed to be added by IrishGen
+(including the ancestor tweak mentioned above) which means less work
+on the part of IrishGen to maintain the special ontologies.
+
+However, the reader may notice that I used a script to add the
+possibility of using Agrelon rather than adding Agrelon directly to
+the IrishGen TRiG files.  There is a formal limitation to using the
+Python script to add Agrelon.  The formal limitation is that Graphs
+have no natural ordering.  This might not seem like an impediment but
+IrishGen's files generally follow the text that they transform.  A
+user should be able to read the IrishGen file and follow along from
+the original source material.  If the script was run and the files
+checked in directly from that, the lexical ordering of the files would
```

Comment:
I'm not sure that "lexical ordering" is the best description. Does "lexical" have a special meaning in this context? If not, "the link between the order of the source text and the ordering of the corresponding file" is my best (if much more verbose) attempt at a description of what we are getting at.